subprojects/base-services/src/main/java/org/gradle/util/internal/TextUtil.java

```diff
@@ -207,4 +207,34 @@ public static String camelToKebabCase(String camelCase) {
     public static String toLowerCaseLocaleSafe(String s) {
         return s.toLowerCase(Locale.ENGLISH);
     }
+
+    /**
+     * Replacement for locale-dependent {@code String.toLowerCase()}.
+     *
+     * <p>In most cases, making a string lowercase should be independent from the current locale (i.e. {@code String.toLowerCase(Locale.XXX)} should be used). If an implementation does need to be
+     * locale-dependent then this method has to be used.
+     *
+     * <p>There's an architecture test in place that will fail if an implementation tries to call {@code String.toLowerCase()} without an argument.
+     *
+     * @param s string to be made lowercase
+     * @return a lowercase string
+     */
+    public static String toLowerCaseUserLocale(String s) {
+        return s.toLowerCase();
+    }
+
+    /**
+     * Replacement for locale-dependent {@code String.toUpperCase()}.
+     *
+     *  <p>In most cases, making a string uppercase should be independent from the current locale (i.e. {@code String.toUpperCase(Locale.XXX)} should be used). If an implementation does need to be
```

Comment:
Same comment as above. Do we also need a toUpperCaseLocaleSafe equivalent?

Reply:
Probably. We have a couple of options here:
- We introduce toUpperCaseLocaleSafe() and use when we feel like
- We enforce `toUpperCaseLocaleSafe()` calls everywhere if favor of `toLowerCase(Locale.ENGLISH)`
- We inline all `toUpperCaseLocaleSafe()` and delete it from `TextUtil`.

WDYT?